rules/type-declaration/src/PHPStan/Type/ObjectTypeSpecifier.php

```diff
@@ -128,7 +128,9 @@ private function matchSameNamespacedObjectType(Node $node, ObjectType $objectTyp
             return null;
         }
 
-        $namespacedObject = $namespaceName . '\\' . $objectType->getClassName();
+        // ltrim: prevents double-backslash that gives fatal error with existing class for class_exists() while
+        // auto-loading #5024
+        $namespacedObject = $namespaceName . '\\' . ltrim($objectType->getClassName(), '\\');
```

Comment:
Does this fix the bug you had?

Could you run first CI without this fix, so we can see the reported bug? Thanks :+1:

Reply:
Yes, this is identified as effective against the fatal error. I'll do the regression test first, this includes auto-loading.

Reply:
CI without this fix is in: https://github.com/rectorphp/rector/pull/5026/commits/88163d6c5f6b315d9649392f2aeb34fa7f3424d7

Phpstan is exceptionally low on anonymous classes and over-estimates itself to not infer the anonymous type (Phpstorm does this). If you know by chance a docblock notation to handle this please let me know, also about suppressing these, I'm not as fluent with Phpstan as you are.

Reply:
Noise reduced by suppression. The one is a known issue with Phpstan (https://github.com/phpstan/phpstan/issues/3144) and the other is an issue with Simplify (just reported: https://github.com/symplify/symplify/issues/2717).